setup_mac_zsh.sh

```diff
@@ -0,0 +1,171 @@
+pip uninstall torch torchvision torchaudio -y
+
+# Normally, we would install the latest nightly build of PyTorch here,
+# But there's currently a performance regression in the latest nightly releases.
+# Therefore, we're going to use this old version which doesn't have it.
+# TODO: go back once fixed on PyTorch side
+pip install --pre torch==1.13.0.dev20220922 torchvision==0.14.0.dev20220924 -f https://download.pytorch.org/whl/nightly/cpu/torch_nightly.html --no-deps
+
+# Missing dependencie(s)
+pip install gdown fastapi
+
+# Activate the MPS_FALLBACK conda environment variable
+conda env config vars set PYTORCH_ENABLE_MPS_FALLBACK=1
+
+# We need to reactivate the conda environment for the variable to take effect
+conda deactivate
+conda activate web-ui
+
+# Check if the config var is set
+if [ -z "$PYTORCH_ENABLE_MPS_FALLBACK" ]; then
+    echo "============================================="
+    echo "====================ERROR===================="
+    echo "============================================="
+    echo "The PYTORCH_ENABLE_MPS_FALLBACK variable is not set."
+    echo "This means that the script will either fall back to CPU or fail."
+    echo "To fix this, please run the following command:"
+    echo "conda env config vars set PYTORCH_ENABLE_MPS_FALLBACK=1"
+    echo "Or, try running the script again."
+    echo "============================================="
+    echo "====================ERROR===================="
+    echo "============================================="
+    exit 1
+fi
+
+# Create a shell script to run the web ui
+echo "#!/usr/bin/env bash -l
+# This should not be needed since it's configured during installation, but might as well have it here.
+conda env config vars set PYTORCH_ENABLE_MPS_FALLBACK=1
+# Activate conda environment
+conda activate web-ui
+# Pull the latest changes from the repo
+git pull --rebase
+# Run the web ui
+python webui.py --precision full --no-half --opt-split-attention-v1 --use-cpu GFPGAN CodeFormer BSRGAN ESRGAN SCUNet
+# Deactivate conda environment
+conda deactivate
+" > run_webui_mac.sh
+
+# Give run permissions to the shell script
+chmod +x run_webui_mac.sh
+
+echo "============================================="
+echo "============================================="
+echo "==============MORE INFORMATION==============="
+echo "============================================="
+echo "============================================="
+echo "If you want to run the web UI again, you can run the following command:"
+echo "./stable-diffusion-webui/run_webui_mac.sh"
+echo "or"
+echo "cd stable-diffusion-webui && ./run_webui_mac.sh"
+echo "============================================="
+echo "============================================="
+echo "============================================="
+echo "============================================="
+
+
+# Run the web UI
+python webui.py --precision full --no-half --opt-split-attention-v1 --use-cpu GFPGAN CodeFormer BSRGAN ESRGAN SCUNet
```

Comment:
You can now remove `--opt-split-attention-v1` from both `python webui.py --precision full --no-half --opt-split-attention-v1 --use-cpu GFPGAN CodeFormer BSRGAN ESRGAN SCUNet` so that the new default, InvokeAI's split attention, gets used. InvokeAI's split attention gives ~30% better performance.

Reply:
Fantastic Improvement. Thank for informing me to update.